runtime/bindings/python/iree/runtime/io.py

```diff
@@ -64,3 +67,137 @@ def save_archive_file(entries: dict[str, Union[Any, SplatValue]], file_path: Pat
         else:
             index.add_buffer(key, value)
     index.create_archive_file(str(file_path))
+
+
+def parameter_index_add_numpy_ndarray(
+    index: ParameterIndex, name: str, array: np.ndarray
+):
+    """Adds an named array to the index."""
+    metadata = _make_tensor_metadata(array)
+    # 0d arrays are special in both torch/numpy in different ways that makes
+    # it hard to reliably get a memory view of their contents. Since we
+    # know that 0d is always small, we just force a copy when in numpy
+    # land and that seems to get it on the happy path.
+    # See: https://github.com/iree-org/iree-turbine/issues/29
+    if len(array.shape) == 0:
+        flat_array = array.copy()
+    else:
+        flat_array = np.ascontiguousarray(array).view(np.uint8)
+    index.add_buffer(name, flat_array, metadata=metadata)
```

Comment:
Oh, I'd nearly forgotten that we have support for arbitrary metadata in parameter archives. So this lets the C runtime load the data as dense bytes without needing to care about the original types, but then Python code can read the metadata to unpack.

On that note, do you want to also write some version information into the metadata, to be able to handle encoding changes more gracefully? Maybe as part of the `_metadata_prefix`, like `NUMPY_v0:`?